Week-1/aryan29/keeptrack.py

```diff
@@ -0,0 +1,67 @@
+import time,json
+import smtplib
+from termcolor import colored,cprint
+import requests
+from datetime import datetime,timedelta
+from api import Api
+dateIndex=datetime.now()
+with open("user_config.json","r") as f:
+    conf=json.load(f)
+
+    if(conf["sender_email"]=="" or conf["sender_password"]=="" or len(conf["receiver_email"])==0):
+        cprint("Please Make sure to Enter Sender_Email, Receiver_Email and Sender_Password for Gmail before continuing",'red')
+        exit(0)
+
+def keeptrack():
+    with open("data.json","r") as f:
+        obj=json.load(f)
+    #print(obj)
+    for mobile in obj:
+        name=mobile['name']
+        oldPrice=mobile['Best Price']
+        newPrice=get_current_price(mobile['Code'])
+        print(oldPrice,newPrice)
+        if(newPrice<oldPrice):
+            message=f"Best Price of {name} has changed from {oldPrice} to {newPrice} at {time.ctime(time.time())}"
+            sendEmail(message)
+            telegram_alert(message)
+            with open("logs.txt","w") as f:
+                print(message,file=f)
+def get_current_price(mob):
+    x=Api().get_data(mob)
+    return Api().name_data(x,mob)['Best Price']
+def sendEmail(msg):
+    try:
+        s = smtplib.SMTP('smtp.gmail.com', 587)
+        s.starttls()
+        s.login(conf["sender_email"], conf["sender_password"])
+        list=conf["receiver_email"]
+        message = "Subject:Price Updates\n\n"+msg
+        print(f"Message is {message}")
+        for lines in list:
+            s.sendmail(conf['sender_email'], lines, message)
+        s.quit()
+    except:
+        cprint("Email Credentials not avilable or they are not valid please allow less secure apps from your gmail if your credentials are correct",'red')
+def telegram_alert(bot_message):
+    try:
+        bot_token=conf['telegram_bot_token']
+        bot_chatID=conf['telegram_chatID']
+        url="https://api.telegram.org/bot"+bot_token+'/sendMessage?chat_id'+bot_chatID+'&parse_mode=Markdown&text='+bot_message
+    except:
+        cprint("Telegram Credentials not avilable or they are not valid",'red')
+    res=requests.get(url)
+def checkserver():
+    msg="Daily Server Check"
+    #telegram_alert(msg)
+    sendEmail(msg)
+if __name__ == "__main__":
+    updation_time=60*60*3 #3 hour updates
+    c=0
+    while(1):
+        c=c+1
+        keeptrack()
+        if(c%8==1):
+            checkserver()    #1day updates
+        time.sleep(updation_time)
+        time.sleep(updation_time)
```

Comment:
I'm not reviewing each line. But these are the general errors I saw after having a look:

- Some lines seem too long to follow PEP8
- Documentation is None.
- Please defer from using multiple imports in a single line.
- For constants, using UPPER_CASE should be the way to go.
- Pick a style.
- Some imports are unused.
- Correct your import order.
- Few variables are unused. e.g.: `res`
- `f` is used in two scopes.

Reply:
Now using PEP8
Import Rules Done
Line Too Long Done
Using mixedCase for Variables, snake_case for functions and UPPER_CASE for constants
Unused variables and functions removed